source/core/src/main/com/csse3200/game/screens/MainGameScreen.java

```diff
@@ -120,25 +121,38 @@ public MainGameScreen(GdxGame game) {
         Stage stage = ServiceLocator.getRenderService().getStage();
         ServiceLocator.registerAlertBoxService(new AlertBoxService(stage, skin));
 
+        // todo load based on what the user chose
+        boolean shouldLoad = gameOptions.shouldLoad;
+        logger.info("Should start from save file: {}", shouldLoad);
+
         /**
          * based on the characters selected, changed the link
          * If Player choose Load, then create
          */
-        // todo confirm which players should be passed into PlayerFactory
-        this.playerFactory = new PlayerFactory(List.of(
-                PLAYERS
-        ));
-        Entity player = playerFactory.createPlayer(
-                FileLoader.readClass(PlayerConfig.class, chosenPlayer).name);
+        this.playerFactory = new PlayerFactory(Arrays.stream(PLAYERS).toList());
+        Entity player;
+        if (shouldLoad) {
+            System.out.println("Should start from save file: {}" + shouldLoad);
```

Comment:
This prints `Should start from save file: {}true`, also line 126 is basically the same message so I don't know why you need this

Reply:
done